gbpservice/nfp/orchestrator/drivers/orchestration_driver.py

```diff
@@ -503,46 +600,40 @@ def create_network_function_device(self, device_data,
             raise exceptions.ComputePolicyNotSupported(
                 compute_policy=device_data['service_details']['device_type'])
 
+        token = device_data['token']
+        admin_tenant_id = device_data['admin_tenant_id']
         image_name = self._get_image_name(device_data)
-        if image_name:
-            self._update_vendor_data(device_data,
-                                 device_data.get('token'))
-        try:
-            interfaces = self._get_interfaces_for_device_create(
-                    device_data,
-                    network_handler=network_handler
-            )
-        except Exception as e:
-            LOG.exception(_LE('Failed to get interfaces for device creation.'
-                              'Error: %(error)s'), {'error', e})
+
+        executor = nfp_executor.TaskExecutor(jobs=3)
+
+        image_id_result = {}
+
+        executor.add_job('UPDATE_VENDOR_DATA',
+                         self._update_vendor_data_v1,
+                         token, admin_tenant_id, image_name, device_data)
+        executor.add_job('GET_INTERFACES_FOR_DEVICE_CREATE',
```

Comment:
While adding job in TaskExecutor, we raise the exception if it is in use,
in that case, we need to handle the exception, else it'll go as unhandled exception to module.